### PR TITLE
Redirect some test spam to files instead of main log files

### PR DESCRIPTION
**Pull request by gmarek**
Ref #44707

@kubernetes/test-infra-maintainers @wojtek-t

### CODE REVIEW

File: test/e2e/framework/framework.go [hunk 1]
```diff
@@ -284,8 +286,25 @@ func (f *Framework) AfterEach() {
...
-		LogContainersInPodsWithLabels(f.ClientSet, metav1.NamespaceSystem, ImagePullerLabels, "image-puller", Logf)
+
+		logFunc := Logf
+		if TestContext.ReportDir != "" {
+			filePath := path.Join(TestContext.ReportDir, "image-puller.txt")
+			By(fmt.Sprintf("Dumping a list of prepulled images on each node to file %v", filePath))
+			file, err := os.Create(filePath)
+			if err != nil {
+				Logf("Failed to create a file with image-puller data %v: %v", filePath, err)
+				return
```
Comment (wojtek-t) [inline test/e2e/framework/framework.go]: Change to else - otherwise you will loose everything that is running later.
[The same you did above]

Comment (gmarek) [inline test/e2e/framework/framework.go]: Done.

**Comment by fejta** [comment]
LGTM

**Comment by cblecker** [comment]
Hmm.. For some reason the "Jenkins GCE e2e" test is showing green on Github, but SQ is reporting it isn't green. Kicking it to see if it'll pick up.
@k8s-bot cvm gce e2e test this

**Pull request merged**